CorsixTH/Lua/dialogs/tree_ctrl.lua

```diff
@@ -622,6 +622,10 @@ function TreeControl:onMouseUp(button, x, y)
       redraw = true
     else
       self.selected_node = node
+      if self.parent and
+          (self.parent.new_savegame_textbox or self.parent.new_map_textbox) then
```

Comment:
no, too leaky. Notice nothing else in this widgit references parent.

Make a callback; try to clearly differentiate from select_callback; you may have to rename that one as well.